business/istio_config.go

```diff
@@ -1020,9 +1020,9 @@ func (in *IstioConfigService) IsAmbientEnabled() bool {
 	}
 	if ambientEnabled == nil || currentTime.Sub(*lastUpdateTime) > time.Minute {
 		ambientEnabled = new(bool)
-		daemonset, err := in.kialiCache.GetDaemonSet(in.config.IstioNamespace, "ztunnel")
+		daemonset, err := in.kialiCache.GetDaemonSet(in.config.ExternalServices.Istio.IstioAmbientNamespace, "ztunnel")
```

Comment:
Is there a label that the ztunnel daemonsets have that we can use to look for them? Rather than doing a `GET` in a specific namespace, can we list across all namespaces using label and just check if `len(result) > 0`? This wouldn't require adding another config option and for the user to set the right config option. You might need to adapt the `kialiCache.GetDaemonSets()` method to work when `accessible_namespaces != **` but just look at the istio config list methods for an example.

Reply:
I see that the istio pod has this annotation:

`Annotations:      ambient.istio.io/redirection: disabled`

At least, at the moment, it looks like the ztunnel namespace is predefined by istio, so, if Kiali knows somehow if we are in OpenShift we can look for the ztunnel in the kube-system namespace. Or even have this 3rd option to change in case ztunnel cannot be found in the defaults namespaces.

Reply:
If this label is stable: `app=ztunnel` I would do something like this: `GetDaemonSets(meta_v1.NamespaceAll, app=ztunnel)`. You'll probably need to adjust the implementation of `GetDaemonSets` or add a new method that accepts a selector label but see one of the istio cache methods for an example: https://github.com/kiali/kiali/blob/757c6b47f41933147e678b7c98f9a0d74ad4f372/kubernetes/cache/kube_cache.go#L1183-L1228

This should work even if `accessible_namespaces != **`.

> At least, at the moment, it looks like the ztunnel namespace is predefined by istio, so, if Kiali knows somehow if we are in OpenShift we can look for the ztunnel in the kube-system namespace. 

I think we should avoid having special cases for Openshift if we can and instead try to find an implementation that works for both.

The `ambient.istio.io/redirection` annotation isn't specific to ztunnel though right? Other daemonsets could also use that annotation? I also don't think you can use annotations in `ListOptions`.

Reply:
Thanks, @nrfox ! This approach works as expected. 

Probably it doesn't need to be checked so often so perhaps I would also considered changing the refresh time: 
https://github.com/kiali/kiali/pull/7162/files#diff-163a601f65fbd474e86f1928ac31ba72659f4fd009c067c6f3226ce066f8861fR1021

What do you think? 

Tested in crc: 
![image](https://github.com/kiali/kiali/assets/49480155/f7f8d402-acf3-49a2-9d63-894c9f936f3c)

Reply:
Yeah 1 minute seems low. 10 minutes?